### ISSUE TITLE
case-insensitivity for `[if]`

### ISSUE THREAD

**Issue by ruv** [albertvanderhorst/ciforth#12]
ciforth version 5.5.1

It seems, `[if]` `[else]` `[then]` are still case-sensitive even though `"CASE-INSENSITIVE" WANTED`.

The command:
```shell
echo '0 [if] 1 . [else] 0 . [then]' | ./lina64 -f ' "AUTOLOAD" WANTED  "CASE-INSENSITIVE" WANTED CR '
```
Prints nothing instead of `0`.

Actual output:
```output
 "AUTOLOAD" WANTED  "CASE-INSENSITIVE" WANTED CR
[if] : (WARNING) NOT PRESENT, THOUGH WANTED
 WANT [if]
 ```

Expected output:
```output
[if] : (WARNING) NOT PRESENT, THOUGH WANTED
 WANT [if]
0
```

**Comment by ruv** [comment]
If ciforth provides the standard word `wordlist`, a [reference implementation](https://forth-standard.org/standard/tools/BracketELSE#contribution-121) for `[if]`, `[else]`, `[then]` can be used.

**Comment by albertvanderhorst** [comment]
It cannot be argued that the supplied solution is non-standard, because [THEN] is not supposed to be a Forth word. It violates the principle of least surprises, though.  All uses of [IF] are a mine field, anyway and I will not recommend it.  I'll put this matter in the FAQ.

**Comment by ruv** [comment]
Yes, if you say that the system supports the case-insensitive mode, the user expects that the words `[if]`, `[else]`, `[then]` can be used in lower case too.  Why not implement this?

**Comment by albertvanderhorst** [comment]
> Op 29-08-2024 15:34 CEST schreef ruv ***@***.***>:
> 
> 
> 
> Yes, if you say that the system supports the case-insensitive mode, the user expects that the words [if], [else], [then] can be used in lower case too. Why not implement this?

I don't want to promote conditional compilation by [IF] where WANTED is more superior, anyway.
I don't subscribe to the notion that everything must be added that is convenient to some. Where Forth has never been designed in the first place, this is a recipee for disaster.
If lina is in case-insensitive mode, words can be found in the dictionary. [THEN] is not a dictionary word. Full stop.
You must get used to it that [THEN] will be recognized in comment and strings. 
See the comment from Bernc Paysan. gforth went into the rabbit hole of ignoring [THEN] in comments, 
but retraced it.

Groetjes Albert

**Comment by ruv** [comment]
> I don't want to promote conditional compilation by `[IF]` where `WANTED` is more superior, anyway.

That's fine, no problem.
JFYI, `WANTED` does not help in portable code when you check what words are available and use different implementations depending on that.  For example, if the system provides `find-name`, you use `find-name`, otherwise you use `find`.


>  `[THEN]` is not a dictionary word

There is a `[THEN]` word. See [15.6.2.2533 `[THEN]`](https://forth-standard.org/standard/tools/BracketTHEN).


> You must get used to it that `[THEN]` will be recognized in comment and strings.

This is irrelevant to the initial problem.

Anyway, yes, you cannot wrap a string literal in `[IF]` `[THEN]` if that string literal contains the substring "[THEN]" or "[IF]". This is by design.  The same problem exists in many programming languages.  For example, in C, you cannot wrap a string literal in a multi-line comment if that string literal contains "*/".  The same in Forth multi-line comments.

**Comment by albertvanderhorst** [comment]
**"ORANGUTAN" WANTED** 
investigates whether the word is available in the current search order.
If not, it seeks to supply it (presumably by a kind of library mechanism.)
No reference to system dependant search mechanism is made.
find FIND FIND-NAME find-name , all those are not available in the ciforth kernel and it is irrelevant for this matter.
After WANTED you get a warning, or the word is available.

**Comment by albertvanderhorst** [comment]
> Op 31-08-2024 16:07 CEST schreef ruv ***@***.***>:
> 
> 
> 
> > I don't want to promote conditional compilation by [IF] where WANTED is more superior, anyway.
> That's fine, no problem.
>  JFYI, WANTED does not help in portable code when you check what words are available and use different implementations depending on that. For example, if the system provides find-name, you use find-name, otherwise you use find.
> > [THEN] is not a dictionary word
> There is a [THEN] word. See 15.6.2.2533 [THEN] (https://forth-standard.org/standard/tools/BracketTHEN).

You got me there. That suggests that [THEN] is parsed regularly and must indeed be affected by case-sensitivity. I will reopen this issue shortly. "[THEN]" or "[Then]" is looked up and what found is to be compared with ' [THEN] . 

Replacing 
      2DUP "[THEN]" COMPARE
with
      FOUND '[THEN] =
